### PR TITLE
feat(core): create noop storage service

### PR DESCRIPTION
**Pull request by xelaint**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

File: libs/core/src/storage/noopstorage/noopstorage.service.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { DaffPersistenceService } from "../persistence.interface";
+
+export class DaffNoopStorageService implements DaffPersistenceService {
```
Comment (griest024) [inline libs/core/src/storage/noopstorage/noopstorage.service.ts:3]: This should be decorated with `@Injectable`.

Comment (xelaint) [reply libs/core/src/storage/noopstorage/noopstorage.service.ts:7]: done

File: libs/core/src/storage/noopstorage/noopstorage.service.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { Injectable } from '@angular/core';
+import { DaffPersistenceService } from '../persistence.interface';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class DaffNoopStorageService implements DaffPersistenceService {
+  setItem(key : string, value: any): void {};
+  getItem(key: string): any {};
```
Comment (griest024) [inline libs/core/src/storage/noopstorage/noopstorage.service.ts:9]: test that this method actually return nothing

File: libs/core/src/storage/noopstorage/noopstorage.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+		expect(service.getItem('testkey')).toEqual(undefined);
+	});
+	
+	it('should not be able to retrieve a value from memory', () => {
+    service.setItem('testkey', 'testvalue');
+
+		expect(service.getItem('testkey')).toEqual(undefined);
+	});
+	
+	it('should be able to delete an item from memory', () => {
```
Comment (griest024) [inline libs/core/src/storage/noopstorage/noopstorage.service.spec.ts:26]: I don't understand this behavior. I would think that a noop service should not be interacting with memory at all.

File: libs/core/src/storage/noopstorage/noopstorage.service.spec.ts [hunk 4]
```diff
@@ -1,11 +1,50 @@
...
```
Comment (damienwebdev) [inline libs/core/src/storage/noopstorage/noopstorage.service.spec.ts:26]: `not` I think is a mistype.

File: libs/core/src/storage/noopstorage/noopstorage.service.spec.ts (hunk 3 above)
Comment (xelaint) [reply libs/core/src/storage/noopstorage/noopstorage.service.spec.ts:26]: @damienwebdev

File: libs/core/src/storage/noopstorage/noopstorage.service.spec.ts (hunk 4 above)
Comment (xelaint) [reply libs/core/src/storage/noopstorage/noopstorage.service.spec.ts]: updated

File: libs/core/src/storage/noopstorage/noopstorage.service.spec.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+	});
+	
+	it('should not be able to delete an item from memory', () => {
+		service.setItem('testkey', 'testvalue');
+		service.removeItem('testkey');
+
+		expect(service.getItem('testkey')).toEqual(undefined);
+	});
+	
+	it('should be able to delete all items from memory', () => {
```
Comment (griest024) [inline libs/core/src/storage/noopstorage/noopstorage.service.spec.ts:33]: you need a `not` here as well

Comment (xelaint) [reply libs/core/src/storage/noopstorage/noopstorage.service.spec.ts]: updated

Comment (xelaint) [reply libs/core/src/storage/noopstorage/noopstorage.service.spec.ts]: updated

**Pull request merged**